mixnet/sim/message.py

```diff
@@ -0,0 +1,42 @@
+import pickle
+from dataclasses import dataclass
+from typing import Self
+
+
+@dataclass
+class Message:
+    """
+    A message structure for simulation, which will be sent through mix nodes
+    and eventually broadcasted to all nodes in the network.
```

Comment:
Is this PR still valid one to review? It looks like it covering the old understanding of dissemination.

Reply:
Yes. This PR measures the dissemination time (mix propagation time from the sender to the last mix) in the full-cycle simulation. But, we are not using this for the queuing mechanism experiments now. Nevertheless, I think it's good to keep this in our code base, so that we can use it when it becomes necessary. As a reviewers, this is the lowest priority (not doesn't need to be reviewed until this becomes necessary).